massgen/tests/test_antigravity_cli_backend.py

```
@@ -0,0 +1,490 @@
+        with patch("shutil.which", return_value=None):
+            home = tmp_path
+            agy_path = home / ".local" / "bin" / "agy"
+            agy_path.parent.mkdir(parents=True)
+            agy_path.write_text("")
+            agy_path.chmod(0o755)
+            monkeypatch.setattr(Path, "home", staticmethod(lambda: home))
+            assert AntigravityCLIBackend._find_agy_cli() == str(agy_path)
+
+    def test_construct_raises_when_binary_missing(self):
+        with patch.object(AntigravityCLIBackend, "_find_agy_cli", return_value=None):
+            with pytest.raises(RuntimeError, match="install.sh"):
```

Ruff (0.15.13)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)